app/crud/company_crud.py

```diff
@@ -0,0 +1,57 @@
+from dateutil.parser import parser
+from sqlalchemy import and_
+from sqlalchemy.orm import Session
+from app.crud.base_crud import CRUDBase
+from app.models.company_model import CompanyModel
+from app.schemas.company_model import CompanyModelCreate, CompanyModelUpdate, CompanyModelBase
+
+
+class CompanyModelCrud(
+    CRUDBase[
+        CompanyModel,
+        CompanyModelCreate,
+        CompanyModelUpdate,
+    ]
+):
+    async def list_by_name(
+            self,
+            db: Session,
+            name: str,
+            ) -> list[CompanyModel]:
+        return db.query(CompanyModel).filter(CompanyModel.name == name).all()
+
+    async def get_list_by_filter(
+            self,
+             db: Session,
+             params: dict,
+             current_page: int,
+             page_size: int,
+             order_by_field: str,
+             order_by: str) -> list[CompanyModel]:
+        data_filter = db.query(CompanyModel)
+
+
+
+        if "name" in params:
```

Comment:
for key , value in params:
   if key in ["name","unitcost"]:
    data_filter = data_filter.filter(getattr(CompanyModel, key) == value)
   if key in []:
     data_filter = data_filter.filter(getattr(CompanyModel, key) like  value)